src/pedlar.test.ts

```diff
@@ -47,12 +47,169 @@ describe('perform()', () => {
     expect(effect).toHaveBeenCalled()
     expect(cleanup).not.toHaveBeenCalled()
   })
+
+  describe('.performAgain()', () => {
+    it("doesn't run the effect again if the dependencies haven't changed", () => {
+      let effect = jest.fn()
+      let result = pedlar.perform(effect, [1])
+      expect(effect).toBeCalledTimes(1)
+
+      result.performAgain([1])
+      expect(effect).toBeCalledTimes(1)
+    })
+
+    it('runs the effect if the dependencies have changed', () => {
+      let effect = jest.fn()
+      let result = pedlar.perform(effect, [1])
+      expect(effect).toBeCalledTimes(1)
+
+      result.performAgain([2])
+      expect(effect).toBeCalledTimes(2)
+
+      result.performAgain([3])
+      expect(effect).toBeCalledTimes(3)
+    })
+
+    it('runs the effect if the dependencies change back to original value', () => {
+      let effect = jest.fn()
+      let result = pedlar.perform(effect, [1])
+      expect(effect).toBeCalledTimes(1)
+
+      result.performAgain([2])
+      expect(effect).toBeCalledTimes(2)
+
+      result.performAgain([1])
+      expect(effect).toBeCalledTimes(3)
+    })
+
+    it('runs the effect if the dependencies have changed to a different type', () => {
+      let effect = jest.fn()
+      let result = pedlar.perform(effect, [1, '2'])
+      result.performAgain(['1', '2'])
```

Comment:
so it appears that you always have to pass in the dependencies to the perform again.. the perform again will just automatically determine if it needs to run a second time based on what was previously stored

Reply:
You do always have to pass dependencies if you have previously passed dependencies.  

But you're exactly right about the second part.  It will determine if the side effect needs to run again based on if the dependencies have changed